packages/api/cms-api/src/public-upload/public-uploads.controller.ts

```diff
@@ -28,4 +40,26 @@ export class PublicUploadsController {
 
         return publicUploadsFile;
     }
+
+    @Get("download/:id")
+    @DisableCometGuards()
+    async downloadFileById(@Param("id") id: string, @Res() res: Response): Promise<void> {
```

Comment:
this adds a public download possibility for any known public upload id. I'm not sure if we really should add that to the library.

Reply:
Do you mean this endpoint should only be available to authenticated users? 
Since they are "public" uploads, it might be necessary, e.g. if the link to a file needs to be shared with someone else.

Reply:
> Since they are "public" uploads, it might be necessary, e.g. if the link to a file needs to be shared with someone else.

Using the name "public upload" was a bad decision, it rather is "non-DAM-uploads".

> Do you mean this endpoint should only be available to authenticated users?

It could be by default. For your use case we could add a backend-for-frontend (BFF) in the site that proxies the call.

Reply:
If these files should not be public, we really need to change the name 😄

Reply:
The endpoint now requires the `publicUploads` permission: f0355b3f396a36d0038e742594546bf67e6fa4f9

Reply:
@nsams we probably need an option to make this public in the module. Otherwise we always require a BFF for all projects

Reply:
> Otherwise we always require a BFF for all projects

In an ideal world the api should not be visible for non-authed-users. And the BFF for public uploads would be a simple nextjs rewrite.

Reply:
> The endpoint now requires the publicUploads permission

and the upload action?

----

i think we should define a few common use-cases where PublicUploads will be used for, and then define an Api that fits for all use-cases.

Reply:
> and the upload action?

I left that, as it was originally, so no permission check. 
Changing this could break usages on existing projects, if they require the endpoint to be public. 
Even if we mention this breaking change in the changelog, I'm not sure how a project would disable the permission check again, if it relies on that.

Reply:
Changed in #2338